src/domain.rs

```diff
@@ -378,7 +392,7 @@ impl<F> Domain<F> {
 
                 // Find all guarded addresses.
                 #[allow(clippy::mutable_key_type)]
-                let mut guarded_ptrs = HashSet::new();
+                let mut guarded_ptrs = Set::new();
```

Comment:
Looking at this code again, I actually wonder if this'd be better as a `Vec` that we just sort and call `uniq` on, especially if the number of hazard pointers is relatively small. But that's a change for another day.